local-notifications/ios/Plugin/LocalNotificationsPlugin.swift

```diff
@@ -540,9 +545,36 @@ public class LocalNotificationsPlugin: CAPPlugin {
     }
 
     func makePendingNotificationRequestJSObject(_ request: UNNotificationRequest) -> JSObject {
-        return [
-            "id": Int(request.identifier) ?? -1
+        var notification: JSObject = [
+            "id": Int(request.identifier) ?? -1,
+            "title": request.content.title,
+            "body": request.content.body
         ]
+
+        var notificationExtras = request.content.userInfo["cap_extra"] as? [String: Any?]
+
+        if notificationExtras == nil {
+            notificationExtras = request.content.userInfo as? [String: Any?]
+        }
+
+        if let notificationExtras = notificationExtras {
+            var extra: JSObject = [:]
+            for(key, value) in notificationExtras {
+                extra[key] = value as? JSValue
+            }
+            notification["extra"] = extra
+        }
+
+        if let notificationSchedule = request.content.userInfo["cap_schedule"] as? [String: Any?] {
+            var schedule: JSObject = [:]
+            for(key, value) in notificationSchedule {
+                schedule[key] = value as? JSValue
+            }
+            notification["schedule"] = schedule
+        }
```

Comment:
Suggested changes as referenced in my comment on https://github.com/ionic-team/capacitor/pull/4133


```suggestion
        let userInfo = JSTypes.coerceDictionaryToJSObject(request.content.userInfo)
        let notificationExtras = userInfo["cap_extra"] as? JSObject ?? userInfo

        if let extra = notificationExtras {
            notification["extra"] = extra
        }

        if let schedule = userInfo["cap_schedule"] as? JSObject {
            notification["schedule"] = schedule
        }
```